pkg/cmd/deletecmd/delete_application.go

```diff
@@ -139,6 +139,23 @@ func (o *DeleteApplicationOptions) Run() error {
 	return nil
 }
 
+func removeEnvironments(jobs []string, environments map[string]*v1.Environment) (applications []string) {
+	var keep bool
+	for _, job := range jobs {
+		keep = true
+		for _, environment := range environments {
+			if strings.Contains(environment.Spec.Source.URL, job) {
+				keep = false
```

Comment:
Why the detour via the `keep`? If you invert the if statement you can do everything in one line, right?

Reply:
@hferentschik I don't think this would work because if it's inverted we'd add records every time we don't match the environment, and that would be the case for every existing environment, so we'd end up with a bunch of duplicates

Reply:
ahh, I see what you are saying.

Reply:
One could split the two for loops, just iterating either the jobs or the envs and then making the "contains" check in a helper function. But that's just nitpick.

Reply:
@hferentschik I've extracted it into a function now ... but I guess this will depend on what we decide about using a different mechanism to get the applications